code/modules/jobs/job_types/other/merc_classes/grenzelhoft.dm

```diff
@@ -3,53 +3,71 @@
-		H.adjust_skillrank(/datum/skill/combat/axesmaces, pick(2,3), TRUE) // Equal chance between skilled and average, can use a cudgel to beat less dangerous targets into submission
-		H.adjust_skillrank(/datum/skill/combat/shields, pick(0,0,1), TRUE)
-		H.adjust_skillrank(/datum/skill/misc/reading, 1, TRUE)
-		H.adjust_skillrank(/datum/skill/craft/cooking, 1, TRUE)
-
-	if(H.gender == FEMALE)
-		H.underwear = "Femleotard"
-		H.underwear_color = CLOTHING_SOOT_BLACK
-		H.update_body()
-
-	beltr = /obj/item/storage/belt/pouch/coins/poor
+/datum/outfit/mercenary/grenzelhoft
 	neck = /obj/item/clothing/neck/chaincoif
 	pants = /obj/item/clothing/pants/grenzelpants
 	shoes = /obj/item/clothing/shoes/rare/grenzelhoft
 	gloves = /obj/item/clothing/gloves/angle/grenzel
 	belt = /obj/item/storage/belt/leather/mercenary
-	beltl = /obj/item/weapon/mace/cudgel
 	shirt = /obj/item/clothing/shirt/grenzelhoft
 	head = /obj/item/clothing/head/helmet/skullcap/grenzelhoft
 	armor = /obj/item/clothing/armor/cuirass/grenzelhoft
-	backl = /obj/item/storage/backpack/satchel
-	backr = /obj/item/weapon/sword/long/greatsword/zwei
-	if(!H.has_language(/datum/language/oldpsydonic))
-		H.grant_language(/datum/language/oldpsydonic)
-		to_chat(H, "<span class='info'>I can speak Old Psydonic with ,m before my speech.</span>")
 
-	H.merctype = 2
+/datum/outfit/mercenary/grenzelhoft/pre_equip(mob/living/carbon/human/H)
+	. = ..()
+	if(H.gender == FEMALE)
+		H.underwear = "Femleotard"
+		H.underwear_color = CLOTHING_SOOT_BLACK
+		H.update_body()
 
-	H.change_stat(STATKEY_STR, 2) // They need this to roll at least min STR for the Zwei.
-	H.change_stat(STATKEY_CON, 2)
-	ADD_TRAIT(H, TRAIT_MEDIUMARMOR, TRAIT_GENERIC)
+/datum/job/advclass/mercenary/grenzelhoft/after_spawn(mob/living/carbon/human/H)
+	. = ..()
+	H.merctype = 2
 	if(H.dna?.species.id == SPEC_ID_HUMEN)
 		H.dna.species.native_language = "Old Psydonic"
 		H.dna.species.accent_language = H.dna.species.get_accent(H.dna.species.native_language)
+	var/weapons = list("Zweihander", "Musket",)
+	var/weapon_choice = input(H,"CHOOSE YOUR WEAPON.", "GO EARN SOME COIN.") as anything in weapons
+	switch(weapon_choice)
+		if("Zweihander")
+			H.equip_to_slot_or_del(new /obj/item/weapon/sword/long/greatsword/zwei, ITEM_SLOT_BACK_R, TRUE)
+			H.equip_to_slot_or_del(new /obj/item/storage/backpack/satchel, ITEM_SLOT_BACK_L, TRUE)
+			H.equip_to_slot_or_del(new /obj/item/storage/belt/pouch/coins/poor, ITEM_SLOT_BELT_R, TRUE)
+			H.equip_to_slot_or_del(new /obj/item/weapon/mace/cudgel, ITEM_SLOT_BELT_L, TRUE)
+			H.adjust_stat_modifier(STATMOD_JOB, STATKEY_STR, 2) // They need this to roll at least min STR for the Zwei.
+			H.adjust_skillrank(/datum/skill/combat/axesmaces, pick(2,3), TRUE) // Equal chance between skilled and average, can use a cudgel to beat less dangerous targets into submission
+			H.adjust_skillrank(/datum/skill/combat/swords, 2, TRUE)
+			if(H.age == AGE_OLD)
+				H.adjust_skillrank(/datum/skill/combat/swords, 1, TRUE)
```

Comment:
We don't want mercs getting master on something even if they are old, it just open a bad precedent